AlphaWallet/Transfer/ViewControllers/SendViewController.swift

```diff
@@ -298,23 +298,36 @@ extension SendViewController: QRCodeReaderDelegate {
         reader.dismiss(animated: true) { [weak self] in
             self?.activateAmountView()
         }
+        guard let result = QRURLParser.from(string: result) else { return }
+        guard checkEIP861Requirements(result: result) else { return }
+    }
 
-        guard let result = QRURLParser.from(string: result) else {
-            return
+    func checkEIP861Requirements(result: ParserResult) -> Bool {
+        //TODO error display on returns
+        //Note: not checking the 'transferType' since erc20 is implied by whether it has uint256 and address.
+        //The contract address can be compared to the one in the token card and if it matches will proceed else fail
+        //this protects the user from sending funds to the wrong address
+        if let chainId = result.params["chainId"] {
+            guard config.chainID == Int(chainId) else { return false }
         }
-        targetAddressTextField.value = result.address
-
-        if let dataString = result.params["data"] {
-            data = Data(hex: dataString.drop0x)
+        //if erc20 (eip861 qr code)
+        if let recipient = result.params["address"], let amt = result.params["uint256"] {
```

Comment:
Should need to check if receipt and amt are empty too.

Reply:
Done